scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -3078,13 +3092,82 @@ static bool antistall_scan(void)
 	return true;
 }
 
+static bool task_count_refresh_dsq(u64 dsq_id, bool reset) {
+	struct task_struct *p;
+	u32 *counter;
+	pid_t pidk;
+	if (!dsq_id)
+		return true;
+	u32 zero_u32 = 0;
+	u32 one_u32 = 1;
+	// verifier
+	bpf_rcu_read_lock();
+	
+	bpf_for_each(scx_dsq, p, dsq_id, 0) {
+		pidk = BPF_CORE_READ(p, tgid);
+		counter = bpf_map_lookup_elem(&tgid_tid_count, &pidk);
+		if (reset) {
+			if (!counter) {
+				bpf_map_update_elem(&tgid_tid_count, &pidk, &zero_u32, 0);
+			} else {
+				*counter = 0;
+			}
+		} else {
+			if (!counter) {
+				bpf_map_update_elem(&tgid_tid_count, &pidk, &one_u32, 0);
+			} else {
+				(*counter)++;
```

Comment:
This is racy. What is this count for? is it for stats? Note that the count can be tracked dynamically from task state transitions - `ops.runnable/running/stopping/quiescent()`.

Reply:
Will set this there instead of via a timer, thanks!

> What is this count for? 

To be able to have policy place spammy tasks into unpriv layer (i.e. for generic things where CommPrefix doesn't work).